tests/integration/remote/invoke/test_remote_invoke.py

```diff
@@ -245,32 +378,46 @@ def setUpClass(cls):
         cls.stack_name = f"{TestNestedTemplateResourcesInvoke.__name__}-{uuid.uuid4().hex}"
         cls.create_resources_and_boto_clients()
 
-    def test_invoke_empty_event_provided(self):
-        command_list = self.get_command_list(
-            stack_name=self.stack_name,
-        )
+    @parameterized.expand(
+        [
+            ("ChildStack/HelloWorldFunction", {"message": "Hello world"}),
+            # Skip remote invoke Step function integration tests
+            # ("ChildStack/HelloWorldStateMachine", "World"),
+        ]
+    )
+    def test_invoke_empty_event_provided(self, resource_id, expected_response):
+        command_list = self.get_command_list(stack_name=self.stack_name, resource_id=resource_id)
 
         remote_invoke_result = run_command(command_list)
         self.assertEqual(0, remote_invoke_result.process.returncode)
         remote_invoke_result_stdout = json.loads(remote_invoke_result.stdout.strip().decode())
-        self.assertEqual(remote_invoke_result_stdout, {"message": "Hello world"})
+        self.assertEqual(remote_invoke_result_stdout, expected_response)
 
-    def test_invoke_with_only_event_provided(self):
+    @parameterized.expand(
+        [
+            ("ChildStack/HelloWorldFunction", '{"key1": "Hello", "key2": "world"}', {"message": "Hello world"}),
+            # Skip remote invoke Step function integration tests
+            # ("ChildStack/HelloWorldStateMachine", '{"key1": "Hello", "key2": "world"}', "World"),
```

Comment:
Just thinking out loud, can we use the dictionary we use in factory implementation to decide whether certain test group should be run or skipped? In that sense we don't need to track to uncomment these lines.

Reply:
Good point, let me think  about changing it to a `skipIf` so that it's easier to maintain.

Reply:
Addressed this in the new commit, used the factory dict so that only that needs to be updated when new resources support is added.